docs/reference/api-docs.asciidoc

```diff
@@ -796,6 +800,83 @@ UpdateStrategy specifies how updates to the cluster should be performed.
 
 
 
+[id="{anchor_prefix}-enterprisesearch-k8s-elastic-co-v1beta1"]
+== enterprisesearch.k8s.elastic.co/v1beta1
+
+Package v1beta1 contains API schema definitions for managing Enterprise Search resources.
+
+.Resource Types
+- xref:{anchor_prefix}-github-com-elastic-cloud-on-k8s-pkg-apis-enterprisesearch-v1beta1-enterprisesearch[$$EnterpriseSearch$$]
+- xref:{anchor_prefix}-github-com-elastic-cloud-on-k8s-pkg-apis-enterprisesearch-v1beta1-enterprisesearchlist[$$EnterpriseSearchList$$]
+
+
+
+[id="{anchor_prefix}-github-com-elastic-cloud-on-k8s-pkg-apis-enterprisesearch-v1beta1-enterprisesearch"]
+=== EnterpriseSearch 
+
+EnterpriseSearch represents an Enterprise Search resource in a Kubernetes cluster.
```

Comment:
Can we make sure to outline that EnterpriseSearch here is a technical way of referring to Enterprise Search, the product? Naming is extra sensitive and we need make sure that we perhaps wrap it in `EnterpriseSearch` a `pre` wrap or something.

Reply:
This doc is auto-generated from the following go code comment:
```go
// EnterpriseSearch represents an Enterprise Search resource in a Kubernetes cluster.
type EnterpriseSearch struct {
```
There's a go linter rule that specifies each commented struct must start with `// <structName> ...`, so it's not easy to change it to eg. `<pre>EnterpriseSearch</pre>`.

I can change it to something like `EnterpriseSearch is a Kubernetes CRD to represent Enterprise Search.` if that makes more sense?

This is the CRD API doc, we'll definitely have more user-facing docs in the future in which it's important we make that distinction clear 👍